### DISCUSSION TITLE
SSG - App Router - when using { cache: 'no-store' } Next no longer generates static HTML files.

### DISCUSSION THREAD

**Discussion by welschmoor** [vercel/next.js#54299] [App Router] [answered]
When adding { cache: 'no-store' } to server side fetch calls, NEXT.js no longer generates static html files.

I want: To prebuild a couple of pages on the server, and build the rest on demand, and at the same time I do not want any of the fetch calls cached. 

Reproduction:

- Clone this official app router demo repo: https://app-router.vercel.app/
- In /app/ssg/[id] add { cache: 'no-store' } as the second parameter to the fetch call.
- Visit pages (http://localhost:3000/ssg/1 ) and look at the .next/server/app/ssg folder and you will see that the static pages are no longer being generated, neither the ones defined inside generateStaticParams. When visiting and revisiting http://localhost:3000/ssg/1 each time it now says "generated 0seconds ago", which means SSG is no longer working.

To circumvent this I tried not using 'no-store' cache headers and played with on demand revalidation and revalidatePath instead without success (it's not revalidating the data that I manually changed in another backend and the fetch is holding on to old data unfortunately). It would be nice having some kind of solution. Update: I got the on demand revalidation to work, apparently I must provide the revalidate fetch option.

Summary: I want to disable the data caching of fetch calls, but I want to keep static generation when using generateStaticParams. I did not want to open an issue, because maybe it's working as intended.

**Comment by oliverkidd** [comment, marked as answer]
Straight from the docs under 'dynamic functions' 

If you don't want caching, you can't have static generation.

<img width="645" alt="image" src="https://github.com/vercel/next.js/assets/124500411/e8238b98-1fdf-499d-a21e-de8f09d2082d">

**Reply by welschmoor** [reply]
Another issue: running yarn build again uses old data for some reason. Only revalidating with api/revalidate gets rid of old data. Even though it seems I see the .next folder being deleted, I still get old data (maybe browser also caches this data? But that would be new to me). I am too tired to look at it today though.

**Reply by alfonsusac** [reply]
Why not? It seems like the ability to cache a fetch should not have anything to do with static generation. Especially if the route is fully static (no revalidation).
